perun-core/src/main/java/cz/metacentrum/perun/core/bl/AttributesManagerBl.java

```diff
@@ -175,6 +175,26 @@
 	List<Attribute> getAttributes(PerunSession sess, Resource resource, Member member, List<String> attrNames, boolean workWithUserAttributes) throws InternalErrorException, WrongAttributeAssignmentException;
 
 	/**
+	 * Gets selected <b>non-empty</b> attributes associated with the member, group and the resource.
```

Comment:
This is not true, when you pass list of attrNames, even empty attributes are retrieved. Non-empty attributes case is only when list of attrNames is empty.